src/main.ts

```diff
@@ -115,7 +115,9 @@ const upload = async (): Promise<void> => {
       throw new Error('Robot id not given');
     }
   } catch (err) {
-    setFailed(err.message);
+    if (err instanceof Error) {
+      setFailed(err.message);
+    }
   }
```

Comment:
We should probably still throw if not an instance of `Error` as we otherwise swallow it:
```
if (err instanceof Error) {
  setFailed(err.message);
}
    
throw err
```

Reply:
Github actions should not fail themselves, so they report the action task fail correctly that is why the `setFailed` callback is used. Updated to report an unknown error in case an `Error` object is not present (it should in most cases, this is more to satisfy Typescript).